include/envoy/http/access_log.h

```diff
@@ -89,9 +91,10 @@ class RequestInfo {
   virtual std::chrono::milliseconds duration() const PURE;
 
   /**
-   * @return the failure reason for richer log experience.
+   * @return the failure reason for richer log experience, this can be represented as binary OR of
+   * FailureReason enum values.
    */
-  virtual FailureReason failureReason() const PURE;
+  virtual uint64_t failureReason() const PURE;
```

Comment:
This name doesn't make sense anymore. I would do the following:

1) Rename FailureReason -> ResponseFlag
2) void setResponseFlag(ResponseFlag flag);
3) bool getResponseFlag(ResponseFlag flag);

This way it is clear what you are adding flags together and can test if a flag is set. In the future we could add a clearResponseFlag() method if we need it.